src/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_factory.js

```diff
@@ -26,10 +27,21 @@ export class VisualizeEmbeddableFactory extends EmbeddableFactory {
   }
 
   render(domNode, panel, container) {
+
     const visualizeScope = this.$rootScope.$new();
     visualizeScope.editUrl = this.getEditPath(panel.id);
     return this.visualizeLoader.get(panel.id)
       .then(savedObject => {
+
+        const labMode = this._config.get('visualize:labMode');
+
+        if (!labMode && savedObject.vis.type.isExperimental) {
+          return new Embeddable({
+            title: `${savedObject.title} is an experimental visualization. 
```

Comment:
I would rather not fit all of this in the title of the dashboard panel. This doesn't fit in the panel in most cases.
Could we rather write this in the body of the panel. My suggestion would be:

```js
if (!labMode && savedObject.vis.type.isExperimental) {
  domNode.innerHTML = `<div class="disabledLabVisualization">
      <div class="kuiVerticalRhythm disabledLabVisualization__icon kuiIcon fa-flask" aria-hidden="true"></div>
      <div class="kuiVerticalRhythm"><em>${savedObject.title}</em> is an experimental visualization.</div>
      <div class="kuiVerticalRhythm">Please turn on experimental mode in the advanced settings to see
      experiemental visualizations.</div>
    </div>`;
  return new Embeddable({
    title: savedObject.title
  });
}
```

and add the following to `src/core_plugins/kibana/public/visualize/styles/main.less` (or better create a `lab_vis.less` in the same folder and import that in the `main.less`):

```less
.disabledLabVisualization {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.disabledLabVisualization__icon {
  font-size: 2em;
}
```

This would provide the following look and feel, which I think would be a bit more beautiful and scalable:

![screenshot-20171114-091249](https://user-images.githubusercontent.com/877229/32769404-0a865d34-c91c-11e7-9228-63f6fab4dadb.png)

Reply:
This above code of course still has the issue, that I would change the description (see in the comment below).